src/abstractdataframe/subset.jl

```diff
@@ -39,12 +31,23 @@ function _and_missing(x::Any...)
                         "but only true, false, or missing are allowed"))
 end
 
-
 # Note that _get_subset_conditions will have a large compilation time
 # if more than 32 conditions are passed as `args`.
 function _get_subset_conditions(df::Union{AbstractDataFrame, GroupedDataFrame},
-                                @nospecialize(args), skipmissing::Bool)
-    conditions = Any[_process_subset_pair(i, a) for (i, a) in enumerate(args)]
+                                wargs::Ref{Any}, skipmissing::Bool)
+    args = only(wargs)
+    conditions = Any[]
+
+    # subset allows a transformation specification without a target column name or a column
+    for (i, a) in enumerate(args)
```

Comment:
Why not use a comprehension?

Reply:
I wanted to eliminate `_proess_subset_pair` function that would get specialized on the types of arguments passed. However, I can do additional timings if this is a significant compilation overhead (probably not as this is a short function).

Reply:
But you can use a comprehension without `_process_subset_pair`, right?

Reply:
but it would be very ugly then as the whole code:
```
        if a isa ColumnIndex
            a => Symbol(:x, i)
        elseif a isa Pair{<:Any, <:Base.Callable}
            first(a) => last(a) => Symbol(:x, i)
        else
            throw(ArgumentError("condition specifier $a is not supported by `subset`"))
        end
```
would have to go into the body of the comprehension.

Reply:
Matter of taste I guess. :-)

Reply:
changed - please have a look how it reads now.